src/rollup/plugins/public-assets.ts

```diff
@@ -112,7 +113,10 @@ export function readAsset (id) {
         const publicAssetBases = Object.fromEntries(
           nitro.options.publicAssets
             .filter((dir) => !dir.fallthrough && dir.baseURL !== "/")
-            .map((dir) => [dir.baseURL, { maxAge: dir.maxAge }])
+            .map((dir) => [
+              withTrailingSlash(dir.baseURL),
+              { maxAge: dir.maxAge },
```

Comment:
I think we can fix it in config resolver instead to make sure `baseURL` always has trailing (https://github.com/unjs/nitro/blob/main/src/core/config/resolvers/assets.ts#L25)

Reply:
I assumed we had a reason for resolving without trailing slash.

https://github.com/unjs/nitro/blob/main/src/core/config/resolvers/assets.ts#L13

For example, in next lines we assume it doesn’t have one (when we generate route rules).

https://github.com/unjs/nitro/blob/main/src/core/config/resolvers/assets.ts#L48

The other issue is if the config is modified subsequently (by, for example a hook), then this issue will reoccur. It feels safer to modify within the plugin.

But of course if you prefer I can update in the config loader/resolver.